src/ziggurat/messaging/consumer.clj

```diff
@@ -40,28 +47,28 @@
         (lb/ack ch delivery-tag))
       (convert-to-message-payload message topic-entity))
     (catch Exception e
-      (lb/reject ch delivery-tag false)
-      (sentry/report-error sentry-reporter e "Error while decoding message")
+      (sentry/report-error sentry-reporter e "Error while decoding message. Publishing it to the dead-letter queue")
       (metrics/increment-count ["rabbitmq-message" "conversion"] "failure" {:topic_name (name topic-entity)})
+      (publish-to-dead-letter topic-entity payload)
       nil)))
 
 (defn- ack-message
   [ch delivery-tag]
   (lb/ack ch delivery-tag))
 
 (defn process-message-from-queue [ch meta payload topic-entity processing-fn]
-  (let [delivery-tag (:delivery-tag meta)
-        message-payload      (convert-and-ack-message ch meta payload false topic-entity)]
+  (let [delivery-tag     (:delivery-tag meta)
+        message-payload  (convert-and-ack-message ch meta payload false topic-entity)]
     (when message-payload
       (log/infof "Processing message [%s] from RabbitMQ " message-payload)
       (try
         (log/debug "Calling processor-fn with the message-payload - " message-payload " with retry count - " (:retry-count message-payload))
         (processing-fn message-payload)
         (ack-message ch delivery-tag)
         (catch Exception e
-          (lb/reject ch delivery-tag true)
-          (sentry/report-error sentry-reporter e "Error while processing message-payload from RabbitMQ")
-          (metrics/increment-count ["rabbitmq-message" "process"] "failure" {:topic_name (name topic-entity)}))))))
+          (sentry/report-error sentry-reporter e "Error while processing message-payload from RabbitMQ. Sending it to dead-letter queue.")
```

Comment:
In case message is sent to dead-letter after an exception, what will happen if the dlq is replayed?

Reply:
it is sent to dead-letter so that the actor developer can fix his code (or fix dependency conflicts) and replay the dlq to re-process the message.

If it can't be processed even after (fixing the code and) replaying, then it is equivalent to a lost message.